students/matt_casali/Lesson06/mailroom_4.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import os, datetime, sys
+now = datetime.datetime.now()
+
+donors = {"William Gates, III": [326892.23, 326892.25], "Mark Zuckerberg": [500.00, 800.00, 2.00], "Jeff Bezos": [877.33],
+          "Paul Allen": [750.23, 23.53, 999.99], "Dakota Dakota": [10.00, 100.00,1000.00]}
+
+def list_donors(database):
+    return ", ".join(x for x in database)
+
+
+def thank_you(database):
+    # Get person's name or provide list.
+    person = input("Who would you like to thank? If you would like a list of donors, enter 'list'. ")
+    if person == "list":
+        print(list_donors(database))
+        person = input("Who would you like to thank? ")
+
+    # Convert the amount to a float and add to donor dictionary
+    try:
+        amount = float(input(f"How much did {person} donate? "))
+    except ValueError:
+        print("You did not enter a numeric value. Please try again.")
+        sys.exit()
+    else:
+        if person not in database:
+            database[person] = [amount]
+        else:
+            database[person] += [amount]
+
+    # Print thank you message
+    directory = input("Please input a directory to save your thank you message. ")
+    try:
+        ty_name = person.replace(" ", "_") + f"_{now.year}_{now.month:0>2d}_{now.day:0>2d}" + ".txt"
+        ty_text = os.path.join(directory, ty_name)
+        f = open(ty_text, "w")
+        f.write(f"Dear {person},\nThank you very much for your donation of ${amount:,.2f}.\nSincerely,\nMatt Casali")
+        f.close()
```

Comment:
use `with` statement here 
https://uwpce-pythoncert.github.io/PythonCertDevel/modules/Files.html#using-files-and-with